File: GetNumberOfK.cpp
/*
题目描述:统计一个数字在排序数组中出现的次数。
*/

#include <vector>

using namespace std;

class Solution {
public:
	int GetNumberOfK(vector<int> data, int k) {
		if (data.empty() || data.size() == 0)
			return 0;
		int left = 0;
		int right = data.size() - 1;
		int mid = 0;
		int res = 0;
		while (left < right)
		{
			mid = (left + right) >> 1;
			if (data[mid] == k)
			{
				res = mid;
				break;
			}
			else if (data[mid] > k)
				right = mid - 1;
			else
				left = mid + 1;
		}
		int total = 0;
		for (int i = res+1; i < data.size(); i++)
			if (data[i] == k)
				total++;
		for (int i = res; i >= 0; i--)
			if (data[i] == k)
				total++;
		return total;
	}
};

File: GetNumberOfK.py
# -*- coding:utf-8 -*-

#题目描述:统计一个数字在排序数组中出现的次数。

class Solution:
	def GetNumberOfK(self, data, k):
		return data.count(k)

	def GetNumberOfK(self,data,k):
		if not data:
			return 0
		low=0
		high=len(data)-1
		cnt=0
		end=len(data)-1
		while low<=high:
			mid=(low+high)/2
			if data[mid]==k:
				while mid>=0 and data[mid]==k:
					mid-=1
				mid+=1
				while mid<=end and data[mid]==k:
					mid+=1
					cnt+=1
				return cnt
			elif data[mid]<k:
				low=mid+1
			else:
				high=mid-1
		return cnt